.github/workflows/tests.yml

```diff
@@ -73,55 +76,57 @@ jobs:
-  #         restore-keys: |
-  #           ${{ runner.os }}-maven-build-
-  #           ${{ runner.os }}-maven-format-
-  #           ${{ runner.os }}-maven-
-  #     - name: Build Project
-  #       env:
-  #         USER_NAME: ${{ secrets.USER_NAME }}
-  #         ACCESS_TOKEN: ${{ secrets.ACCESS_TOKEN }}
-  #       run: |
-  #         BUILD="mvn -s $GITHUB_WORKSPACE/.github/workflows/settings.xml -V -B -e -Pdev,examples,assemble,spotbugs -Dmaven.build.cache.enabled=false -DskipServices -Ddeploy -Ddist -T1C -pl "-:config-service" clean install -DskipTests"
-  #         $BUILD
-  #     - name: Run Microservice Unit Tests
-  #       env:
-  #         USER_NAME: ${{ secrets.USER_NAME }}
-  #         ACCESS_TOKEN: ${{ secrets.ACCESS_TOKEN }}
-  #       run: |
-  #         RUN_TESTS="mvn -s $GITHUB_WORKSPACE/.github/workflows/settings.xml -V -B -e -Dmaven.build.cache.enabled=false verify"
-  #         cd microservices
-  #         $RUN_TESTS \
-  #           || { echo "***** TESTS FAILED. Attempting retry."; $RUN_TESTS; } \
-  #           || { echo "***** TESTS FAILED. Attempting final retry."; $RUN_TESTS; }
+  # Build the code and run the microservice unit/integration tests.
+  build-and-test-microservices:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Code
+        uses: actions/checkout@v3
+        with:
+          submodules: 'recursive'
+      - name: Set up JDK ${{env.JAVA_VERSION}}
+        uses: actions/setup-java@v3
+        with:
+          distribution: ${{env.JAVA_DISTRIBUTION}}
+          java-version: ${{env.JAVA_VERSION}}
+      - uses: actions/cache@v3
+        with:
+          path: ~/.m2/repository
+          key: ${{ runner.os }}-maven-build-${{ hashFiles('**/pom.xml') }}
+          restore-keys: |
+            ${{ runner.os }}-maven-build-
+            ${{ runner.os }}-maven-format-
+            ${{ runner.os }}-maven-
+      - name: Build Project
+        env:
+          USER_NAME: ${{ secrets.USER_NAME }}
+          ACCESS_TOKEN: ${{ secrets.ACCESS_TOKEN }}
+        run: |
+          BUILD="mvn -s $GITHUB_WORKSPACE/.github/workflows/settings.xml -V -B -e -Pdev,examples,assemble,spotbugs -DskipServices -Dmaven.build.cache.enabled=false -Ddeploy -Ddist -T1C -pl "-:config-service" clean install -DskipTests"
+          $BUILD
+      - name: Run Microservice Unit Tests
+        env:
+          USER_NAME: ${{ secrets.USER_NAME }}
+          ACCESS_TOKEN: ${{ secrets.ACCESS_TOKEN }}
+        run: |
+          RUN_TESTS="mvn -s $GITHUB_WORKSPACE/.github/workflows/settings.xml -V -B -e verify"
+          cd microservices
+          $RUN_TESTS \
+            || { echo "***** TESTS FAILED. Attempting retry."; $RUN_TESTS; } \
+            || { echo "***** TESTS FAILED. Attempting final retry."; $RUN_TESTS; }
 
   quickstart-build-and-test:
     runs-on: ubuntu-latest
     steps:
     - name: Checkout Code
       uses: actions/checkout@v3
+      with:
+        submodules: 'recursive'
```

Comment:
This is useful while this exists as a PR, but I will remove this once the PR is merged.